src/App.tsx

```diff
@@ -14,11 +15,99 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const minLength = 1;
+  const [isStarted, setIsStarted] = useState(false);
+  const [lengthValue, setLengthValue] = useState(minLength);
+  const [goods, setGoods] = useState([...goodsFromServer]);
+
+  const filteredGoods = [...goods].filter(good => good.length >= lengthValue);
+
+  const reset = () => {
+    setGoods([...goodsFromServer]);
+    setLengthValue(minLength);
+  };
+
+  const reverse = () => {
+    setGoods([...goods].reverse());
+  };
+
+  const sortByAlphabet = () => {
+    setGoods([...goods].sort((g1, g2) => g1.localeCompare(g2)));
+  };
+
+  const sortByLength = () => {
+    setGoods([...goods].sort((g1, g2) => g1.length - g2.length));
+  };
+
+  return (
+    <div className="App">
+      {!isStarted ? (
+        <button
+          type="button"
+          onClick={() => setIsStarted(!isStarted)}
+          className="start-button"
+        >
+          Start
+        </button>
+      ) : (
+        <div className="goods">
+          <GoodsList goods={filteredGoods} />
+          <div className="goods__buttons">
+            <button
+              type="button"
+              onClick={reverse}
+              className="goods__button"
+            >
+              Reverse
+            </button>
+
+            <button
+              type="button"
+              onClick={sortByAlphabet}
+              className="goods__button"
+            >
+              Sort alphabetically
+            </button>
+
+            <button
+              type="button"
+              onClick={reset}
+              className="goods__button"
+            >
+              Reset
+            </button>
+
+            <button
+              type="button"
+              onClick={sortByLength}
+              className="goods__button"
+            >
+              Sort by length
+            </button>
+          </div>
+
+          <form className="goods__filter">
+            <p>Filter by length:</p>
+            <select
+              onChange={(e) => setLengthValue(+e.target.value)}
+              defaultValue={lengthValue}
+            >
```

Comment:
if you change the length to 4 and click reset, your list will reset but on select will be 4